app/util.py

```diff
@@ -11,8 +11,61 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import re
+from dateparser import DateDataParser
+from functools import lru_cache
+
+parser = DateDataParser(languages=['en'])
 
 def canonical_wikilink(wikilink):
+
+    if is_journal(wikilink):
+        try:
+            wikilink = canonical_date(wikilink)
+        except:
+            # TODO: if we add logging, maybe log that we couldn't parse a date here
+            pass
+
     # hack hack
-    wikilink = wikilink.lower().replace(' ', '-').replace('\'', '').replace(',', '')
+    wikilink = (
+        wikilink.lower()
+        .replace(' ', '-')
+        .replace('\'', '')
+        .replace(',', '')
+        .replace('/', '-')
+    )
     return wikilink
+
+
+@lru_cache(maxsize=None)
+def canonical_date(wikilink):
+    date = parser.get_date_data(wikilink).date_obj
+    try:
+        new_wikilink = date.isoformat().split("T")[0] 
+    except:
+        pass
+
+    return new_wikilink
```

Comment:
This will blow up if there's an exception previously -- it seems it'd be better to default new_link to wikilink?

Reply:
thanks, good catch. just pushed up a fix